presto-bigquery/src/main/java/com/facebook/presto/plugin/bigquery/BigQuerySplit.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.bigquery;
+
+import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.spi.ConnectorSplit;
+import com.facebook.presto.spi.HostAddress;
+import com.facebook.presto.spi.schedule.NodeSelectionStrategy;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.facebook.presto.spi.schedule.NodeSelectionStrategy.NO_PREFERENCE;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class BigQuerySplit
+        implements ConnectorSplit
+{
+    private static final int NO_ROWS_TO_GENERATE = -1;
+
+    private final String streamName;
+    private final String avroSchema;
+    private final List<ColumnHandle> columns;
+    private final long emptyRowsToGenerate;
+
+    @JsonCreator
+    public BigQuerySplit(
+            @JsonProperty("streamName") String streamName,
+            @JsonProperty("avroSchema") String avroSchema,
+            @JsonProperty("columns") List<ColumnHandle> columns,
+            @JsonProperty("emptyRowsToGenerate") long emptyRowsToGenerate)
+    {
+        this.streamName = requireNonNull(streamName, "streamName cannot be null");
+        this.avroSchema = requireNonNull(avroSchema, "avroSchema cannot be null");
+        this.columns = ImmutableList.copyOf(requireNonNull(columns, "columns cannot be null"));
+        this.emptyRowsToGenerate = emptyRowsToGenerate;
+    }
+
+    static BigQuerySplit forStream(String streamName, String avroSchema, List<ColumnHandle> columns)
```

Comment:
this means it's an non-empty projection, read stream returns rows from bigquery storage without intermediary.

Reply:
I have the same concern with @qqibrow the definition of emptyRowsToGenerate NO_ROWS_TO_GENERATE are misleading. After we merge the EmptySplitPageSource class, our logic could be simpler. 
Let's fix this class after merge the EmptySplitPageSource class.

Reply:
@qqibrow @highker @beinan what do you think? I feel keeping EmptySplitPageSource makes the logic complicated, this NO_ROWS_TO_GENERATE is misleading. Do you think we could keep EmptySplitPageSource for now? Or we need to merge it in this commit?

Reply:
I'm also not feeling good about NO_ROWS_TO_GENERATE.  But as @fgwang7w  insisted repeatedly, I guess it might be not an easy work to replace it in this commit.  keeping it or replacing it, I'm good either way, your call.